File: liyandong/Kotlin/objectAnimator/src/main/java/com/example/objectanimatordemo/MainActivity.java
package com.example.objectanimatordemo;

import androidx.appcompat.app.AppCompatActivity;

import android.animation.ObjectAnimator;
import android.graphics.Point;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import com.example.objectanimatordemo.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding inflate;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        inflate = ActivityMainBinding.inflate(LayoutInflater.from(this));
        setContentView(inflate.getRoot());
        inflate.btnStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                objAnimatorAlpha();
            }
        });
        inflate.btnStart2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                objAnimatorRotation();
            }
        });
        inflate.btnStart3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                objAnimatorTranslationX();
            }
        });
        inflate.btnStart4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                objAnimatorScaleY();
            }
        });
        inflate.btnStart5.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                objAnimatorFallingBall();
            }
        });
    }

    private void objAnimatorFallingBall() {
        ObjectAnimator alpha = ObjectAnimator.ofObject(inflate.tv5, "FallingPos",
                new FallingBallEvaluator(),
                new Point(0, 0), new Point(400, 400));
        alpha.setDuration(1000 * 2);
        alpha.start();
    }


    //透明度
    private void objAnimatorAlpha() {
        ObjectAnimator alpha = ObjectAnimator.ofFloat(inflate.tv, "alpha", 1, 0, 1);
        alpha.setDuration(2000);
        alpha.start();
    }

    //旋转
    private void objAnimatorRotation() {
        ObjectAnimator alpha = ObjectAnimator.ofFloat(inflate.tv2, "rotation", 0, 270, 0);
        alpha.setDuration(1000 * 8);
        alpha.start();
    }

    //x轴平移
    private void objAnimatorTranslationX() {
        ObjectAnimator alpha = ObjectAnimator.ofFloat(inflate.tv3, "translationX", 0, 200, -200, 0);
        alpha.setDuration(1000 * 4);
        alpha.start();
    }

    //y轴缩放
    private void objAnimatorScaleY() {
        ObjectAnimator alpha = ObjectAnimator.ofFloat(inflate.tv4, "scaleY", 0, 4, 1);
        alpha.setDuration(1000 * 4);
        alpha.start();
    }
}

File: tianxiaolong/sample/ViewBindingSmaple/app/src/main/java/com/location/viewbindingsmaple/MainActivity.kt
package com.location.viewbindingsmaple

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.widget.Button
import android.widget.Toast
import com.location.viewbindingsmaple.databinding.ActivityMainBinding
import kotlinx.android.synthetic.main.activity_main_2.*

class MainActivity : AppCompatActivity() {
    lateinit var  binding:ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(LayoutInflater.from(this))
        setContentView(binding.root)
        //使用viewBinding 获取View
        binding.testBtn.setOnClickListener {
            Toast.makeText(this,"测试view",Toast.LENGTH_SHORT)
        }
        //使用Kotlin方式获取view
        //kotlin可以获取任意布局的view 但是如果当前Activity fragment 设置的不是当前view 就会崩溃 不安全
        //下方代码会奔溃
//        test_2_btn.setOnClickListener {
//            Toast.makeText(this,"崩溃view",Toast.LENGTH_SHORT)
//        }

    }
}
